wemake_python_styleguide/visitors/ast/statements.py

```diff
@@ -149,9 +154,56 @@ def visit_statement_with_body(self, node: StatementWithBody) -> None:
         if isinstance(node, ast.Try):
             self._check_internals(node.finalbody)
 
+        self._check_swapped_variables(node.body)
         self._check_useless_node(node, node.body)
         self.generic_visit(node)
 
+    def _check_swapped_variables(
+        self,
+        body: Sequence[ast.stmt],
+    ) -> None:
+        for assigns in sequence_of_node((ast.Assign,), body):
+            self._almost_swapped(
+                cast(Sequence[ast.Assign], assigns),
```

Comment:
Do we need a `cast` here? Why can't we return the correct type from `sequence_of_node`?

Reply:
done